File: statsig-php/tests/MockPersistentStorage.php
<?php

namespace Statsig\Tests;

use Statsig\PersistentStorage;
use Statsig\StickyValues;

class MockPersistentStorage extends PersistentStorage
{
    public $loadCalls = [];
    public $saveCalls = [];
    public $deleteCalls = [];
    private $storage = [];

    public function load(string $key): ?array
    {
        $this->loadCalls[] = $key;
        return $this->storage[$key] ?? null;
    }

    public function save(string $key, string $config_name, StickyValues $data): void
    {
        $this->saveCalls[] = [
            'key' => $key,
            'config_name' => $config_name,
            'data' => $data
        ];

        if (!isset($this->storage[$key])) {
            $this->storage[$key] = [];
        }
        $this->storage[$key][$config_name] = $data->toArray();
    }

    public function delete(string $key, string $config_name): void
    {
        $this->deleteCalls[] = [
            'key' => $key,
            'config_name' => $config_name
        ];

        if (isset($this->storage[$key][$config_name])) {
            unset($this->storage[$key][$config_name]);
        }
    }
}

File: statsig-dotnet/test/MockPersistentStorage.cs
using System.Collections.Generic;

namespace Statsig.Tests
{
    public class MockPersistentStorage : PersistentStorage
    {
        public IList<string> LoadCalls { get; } = [];
        public IList<SaveCall> SaveCalls { get; } = [];
        public IList<DeleteCall> DeleteCalls { get; } = [];
        private readonly Dictionary<string, Dictionary<string, StickyValues>> storage = [];

        public override IDictionary<string, StickyValues> Load(string key)
        {
            LoadCalls.Add(key);
            return storage.TryGetValue(key, out var values)
                ? new Dictionary<string, StickyValues>(values)
                : [];
        }

        public override void Save(string key, string config_name, StickyValues data)
        {
            SaveCalls.Add(new SaveCall(key, config_name, data));
            if (!storage.TryGetValue(key, out var configs))
            {
                configs = [];
                storage[key] = configs;
            }

            configs[config_name] = data;
        }

        public override void Delete(string key, string config_name)
        {
            DeleteCalls.Add(new DeleteCall(key, config_name));
            if (storage.TryGetValue(key, out var configs))
            {
                configs.Remove(config_name);
                if (configs.Count == 0)
                {
                    storage.Remove(key);
                }
            }
        }

        public sealed record SaveCall(string Key, string ConfigName, StickyValues Data);

        public sealed record DeleteCall(string Key, string ConfigName);
    }
}
